operator/data/examples/multicluster/values-istio-multicluster-primary.yaml

```diff
@@ -14,8 +14,7 @@ spec:
       meshNetworks:
         network1:
           endpoints:
-          # Always use Kubernetes as the registry name for the main cluster in the mesh network configuration
-          - fromRegistry: Kubernetes
+          - fromRegistry: main0
```

Comment:
I think I'm nervous with such a change, as it is different than what we tested in the past few releases.  I think it was intentional to use Kubernetes as we don't want to look up remote cluster's gateway ip for endpoints in the local cluster.

Reply:
The main problem is [that we rely on `fromRegistry` to match the clusterID when initializing the kube Controller](https://github.com/istio/istio/blob/f584ba92977e48d0d6b173c6e2d80ba511a701fe/pilot/pkg/serviceregistry/kube/controller/controller.go#L1074). 

I think what's happening that the [default](https://github.com/istio/istio/blob/f584ba92977e48d0d6b173c6e2d80ba511a701fe/pilot/pkg/features/pilot.go#L298) clusterID of "Kubernetes" is being overridden, so the "networkForRegistry" value isn't set on the Controller.

Reply:
I've backtested to 1.5 and this issue with cluster-local exists there as well.

Reply:
I'm not sure how this would cause us to use the gateway for endpoints in the local cluster? It sets network1 (the primary network) to use its own cluster.

Reply:
Ok, this starts to make sense to me.  Can you explain to me why clusterID is being overridden now because of cluster-local?

Reply:
Also, how would you differentiate local cluster vs remote cluster if we just take the cluster name as the value for `fromRegistry`?

Reply:
cluster-local doesn't cause it to be overridden, the chain goes:
1. The cluster-id of pilot is overridden to be whatever is set in values.global...
2. We perform the initNetworkLookup in the kube Controller registry. In there we check the meshNetworks config for an endpoint matching the cluster name, then set the `networkForRegistry` when we find a network that points to this cluster. 
3. When we configure Kubernetes it won't match with the cluster ID, so `networkForRegistry` never gets set
4. As part of initProxy, we use this to find the cluster that the proxy lives on. I'm not sure whether or not that's correct; it feels sort of wrong. Because the networkForRegistry value isn't set, we won't find the matching cluster for the proxy. 
5. ClusterID isn't set on proxies, so when we check IsClusterLocal we're checking `"the-local-cluster-name" == ""`

Reply:
@stevenctl Agree with your point 1, 2, 3.  I had always thought we don't need to set the networkForRegistry when cluster is local and has the local registry as there is no need to modify the endpoinst when the pods actually run in the local cluster.  The configuring  `Kubernetes` will help in `skipSearchingRegistryForProxy()` to quickly tell istiod if the registry should be searched or not.  Also configuring registry to Kubernetes saves from the need to determine which registries it is since our code has 4 types of registry providers (Mock, K8S, Consul, MCP).